src/App.jsx

```diff
@@ -1,11 +1,14 @@
 import Header from "./components/Header";
 import Feed from "./components/Feed";
+import { useState } from "react";
 
 function App() {
+  const [topic, setTopic] = useState("All Topics");
```

Comment:
Try not to have any states in the highest level of your app - any changes here in state causes your entire app to re-render - it may make more sense to move the topic setting functionality to your articles component - this would also mean that you no longer need the 'Feed' component. Your routes can be brought directly into App, and you can fetch your articles within the articles component and declare the topics state in there too, and fetch your single article within the ArticleByID component. this would reduce the complexity of your app and reduce the number of re-renders caused by states being declared higher up within your app than they need to be. It would also mean your component tree structure would follow a more conventional layout due to a better separation of concerns. Please ask me if that didn't make sense or needs clarification!